testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java

```diff
@@ -6222,6 +6222,49 @@ void checkRegexpExtract(SqlOperatorFixture f0, FunctionAlias functionAlias) {
     f.checkNull("log(10, cast(null as real))");
   }
 
+  /** Test case for
+   * <a href="https://issues.apache.org/jira/browse/CALCITE-6224">[CALCITE-6224]
+   * Add LOG2 function (enabled in MYSQL, Spark library)</a>. */
+  @Test void testLog2Func() {
+    final SqlOperatorFixture f0 = fixture();
+    f0.checkFails("^log2(4)^",
+        "No match found for function signature LOG2\\(<NUMERIC>\\)", false);
+    final Consumer<SqlOperatorFixture> consumer = f -> {
+      f.setFor(SqlLibraryOperators.LOG2);
+      f.checkScalarApprox("log2(2)", "DOUBLE NOT NULL",
+          isWithin(1.0, 0.000001));
+      f.checkScalarApprox("log2(4)", "DOUBLE NOT NULL",
+          isWithin(2.0, 0.000001));
+      f.checkScalarApprox("log2(65536)", "DOUBLE NOT NULL",
+          isWithin(16.0, 0.000001));
+      f.checkScalarApprox("log2(2.0/3)", "DOUBLE NOT NULL",
+          isWithin(-0.5849625007211561, 0.000001));
+      f.checkScalarApprox("log2(4.0/3)", "DOUBLE NOT NULL",
+          isWithin(0.4150374992788435, 0.000001));
+      f.checkScalarApprox("log2(0.5)", "DOUBLE NOT NULL",
+          isWithin(-1.0, 0.000001));
+      f.checkScalarApprox("log2(cast(10e8 as double))", "DOUBLE NOT NULL",
+          isWithin(29.897352853986263, 0.000001));
+      f.checkScalarApprox("log2(cast(10e8 as float))", "DOUBLE NOT NULL",
+          isWithin(29.897352853986263, 0.000001));
+      f.checkScalarApprox("log2(1e+52)", "DOUBLE NOT NULL",
```

Comment:
I added all the negative tests I could think of. At present, the test results are consistent with mysql.

Reply:
Which of these are negative tests?

Reply:
log2(1e+52) mainly wants to simulate the situation of log2(-Infinity), as introduced on the mysql official website

Reply:
[mihaibudiu](https://github.com/mihaibudiu) As mihaibudiu mentioned, I was worried that +0.0 and -0.0 might be different in mysql, so I also tested it in mysql and returned null. I also added the corresponding test in calcite.

Reply:
1e+52 should be converted to +infinity.

Reply:
I'm not sure whether 1e+52 will be converted into infinity in mysql, but as far as the results are concerned, currently the result of log2(1e+52) corresponds to mysql

Reply:
I want to simulate infinity through some methods, such as pow(0,3) but this obviously fails in mysql

Reply:
Maybe what you said makes sense, but I don't plan to solve it in this jira, because this may not only be a problem with the log function, but many Math functions may have similar problems. I will check the practices of other databases and mention one. Jira or email to discuss it

Reply:
I tested mysql's log(+infinity) and returned NULL. .Postgres's log(+infinity) returns infinity, which will report an error in calcite. It seems to be a bug in calcite.
```sql
select log10('-Infinity');
```

Reply:
It seems that calcite cannot convert infinity into the correct java type

Reply:
I think the infinity type should not be fixed in this jira. We discussed a lot about adapting the log2 function and I learned a lot. But at present, infinite adaptation does not seem to be a simple problem, or even just a problem of the log function. I will send an email or jira to discuss it.